Maven apps broken after introducing Gradle Catalog Version

### Expected Behavior

Maven apps should work.

### Actual Behaviour

Maven apps doesn't work when using Micronaut 3.0.2-SNAPSHOT that includes the new Gradle Catalog version.

```
$ mw mn:run    
[INFO] Scanning for projects...
Downloading from sonatype-snapshots: https://s01.oss.sonatype.org/content/repositories/snapshots/io/micronaut/micronaut-parent/3.0.2-SNAPSHOT/maven-metadata.xml
Downloading from central: https://repo.maven.apache.org/maven2/io/micronaut/micronaut-parent/3.0.2-SNAPSHOT/maven-metadata.xml
Downloaded from sonatype-snapshots: https://s01.oss.sonatype.org/content/repositories/snapshots/io/micronaut/micronaut-parent/3.0.2-SNAPSHOT/maven-metadata.xml (605 B at 852 B/s)
Downloading from sonatype-snapshots: https://s01.oss.sonatype.org/content/repositories/snapshots/io/micronaut/micronaut-parent/3.0.2-SNAPSHOT/micronaut-parent-3.0.2-20210908.165331-8.pom
Downloaded from sonatype-snapshots: https://s01.oss.sonatype.org/content/repositories/snapshots/io/micronaut/micronaut-parent/3.0.2-SNAPSHOT/micronaut-parent-3.0.2-20210908.165331-8.pom (11 kB at 40 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/io/micronaut/micronaut-bom/3.0.2-SNAPSHOT/maven-metadata.xml
Downloading from sonatype-snapshots: https://s01.oss.sonatype.org/content/repositories/snapshots/io/micronaut/micronaut-bom/3.0.2-SNAPSHOT/maven-metadata.xml
Downloaded from sonatype-snapshots: https://s01.oss.sonatype.org/content/repositories/snapshots/io/micronaut/micronaut-bom/3.0.2-SNAPSHOT/maven-metadata.xml (952 B at 3.9 kB/s)
Downloading from sonatype-snapshots: https://s01.oss.sonatype.org/content/repositories/snapshots/io/micronaut/micronaut-bom/3.0.2-SNAPSHOT/micronaut-bom-3.0.2-20210908.165331-8.pom
Downloaded from sonatype-snapshots: https://s01.oss.sonatype.org/content/repositories/snapshots/io/micronaut/micronaut-bom/3.0.2-SNAPSHOT/micronaut-bom-3.0.2-20210908.165331-8.pom (52 kB at 142 kB/s)
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[ERROR] Invalid packaging for parent POM io.micronaut:micronaut-bom:3.0.2-SNAPSHOT, must be "pom" but is "toml" @ io.micronaut:micronaut-bom:3.0.2-SNAPSHOT, /home/ivan/.m2/repository/io/micronaut/micronaut-bom/3.0.2-SNAPSHOT/micronaut-bom-3.0.2-SNAPSHOT.pom, line 13, column 14
 @ 
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]   
[ERROR]   The project example.micronaut:maven:0.1 (/home/ivan/workspaces/micronaut-graal-tests/micronaut-maven-graal/pom.xml) has 1 error
[ERROR]     Invalid packaging for parent POM io.micronaut:micronaut-bom:3.0.2-SNAPSHOT, must be "pom" but is "toml" @ io.micronaut:micronaut-bom:3.0.2-SNAPSHOT, /home/ivan/.m2/repository/io/micronaut/micronaut-bom/3.0.2-SNAPSHOT/micronaut-bom-3.0.2-SNAPSHOT.pom, line 13, column 14
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
```

### Steps To Reproduce

- `git clone https://github.com/micronaut-graal-tests/micronaut-maven-graal`
- `cd micronaut-maven-graal`
- `git checkout 3.0.x`
- `./mvnw mn:run`


### Environment Information

Linux & JDK 8

### Example Application

_No response_

### Version

3.0.2-SNAPSHOT

--- a/bom/build.gradle
+++ b/bom/build.gradle
@@ -62,7 +62,9 @@ publishing {
         publication(MavenPublication) {
             artifactId("micronaut-bom")
             from components.javaPlatform
-
+            pom {
+                packaging 'pom'
+            }
             libsCatalog.versionAliases.each { alias ->
                 if (alias.startsWith('managed.')) {
                     libsCatalog.findVersion(alias).ifPresent { version ->

